File: sisuAnalise/src/main/java/sisu/CalculadoraNotaCorte.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package sisu;

import java.util.*;
import java.util.stream.Collectors;

/**
 *
 * @author pc
 */
public class CalculadoraNotaCorte {
    public static Map<String, Map<Integer, Double>> calcularNotaCorte(List<Candidato> candidatos, String curso){
        Map<String, Map<Integer, Double>> resultado = new HashMap<>();
        
        for(Candidato c : candidatos){
            if(!c.getCurso().equalsIgnoreCase(curso)) continue;
            
            String campus = c.getCampus();
            int ano = c.getAno();
            double media = c.getAno();
            
            resultado.putIfAbsent(campus, new HashMap<>());
            Map<Integer, Double> porAno = resultado.get(campus);
            
            if(!porAno.containsKey(ano) || media < porAno.get(ano)){
                porAno.put(ano, media);
            }
        }
        return resultado;
    }
    
    public static Map<String, Double> calcularNotaCorte(List<Candidato> candidatos, boolean fcampus){
        Map<String, Double> resultado = new HashMap<>();
        
        for(Candidato c : candidatos){
            String curr = c.getCurso();
            if(!fcampus) curr += (" - " + c.getCampus());
            
            if(resultado.containsKey(curr)){
                resultado.put(curr, Math.min(c.getMedia(), resultado.get(curr)));
            }else resultado.put(curr, c.getMedia());
        }
        return resultado;
    }
}

File: sisuPy/calculadora_nota_corte.py

from typing import List, Dict
from candidato import Candidato

class CalculadoraNotaCorte:
    def calcular(candidatos: List[Candidato], fcampus: bool) -> Dict[str, float]:
        resultado = {}

        for c in candidatos or []:
            curso = c.curso
            campus = c.campus

            try:
                media_val = float(c.media)
            except Exception:
                continue

            chave = curso if fcampus else f"{curso} - {campus}"

            if chave in resultado:
                # mantém o menor valor (nota de corte = mínimo)
                resultado[chave] = min(media_val, resultado[chave])
            else:
                resultado[chave] = media_val
        return dict(sorted(resultado.items()))